CONTRIBUTING.md

```diff
@@ -0,0 +1,35 @@
+tracker, in particular those marked "good first issue". If you have found a bug, check first that we do not have an existing issue covering it, and if not, feel free to create an issue.
+
+Please follow these guidelines when contributing:
+
+- Ask to be assigned to an issue you wish to work on first (we have lots of past issues, some of which are no longer relevant or wanted);
+- We ask you only to pick issues for which you are confident that you have a solution;
+- If you are a newcomer to the repo, create a personal fork and clone it locally. Check out a new branch with a name relevant to the issue you are working on;
+- Follow the coding style of the area you are editing, including indentation, and be consistent with quotation marks and spacing;
+- Use no higher than [ECMAScript 5](https://caniuse.com/es5) - notably, do not use arrow functions or `async` functions. However, Promises *are*
+  supported via a polyfill;
+- Do not prettify code you are not working on;
+- You must test your code yourself before asking for review, like this:
+  - set up a local Web server (we recommend Node/NPM's [http-server](https://www.npmjs.com/package/http-server));
+  - serve the root directory of the repository (e.g. `http-server .`);
+  - in a browser, navigate to the URL of the main `index.html` (e.g. http://localhost:8080/www/index.html);
+  - manually test your fix in at least Firefox and Chromium (Edge or Chrome), ideally also in IE11 or in "IE Mode" in Edge;
+  - be sure to test your fix in both "JQuery" mode and "Service Worker" mode (see Configuration);
+  - run the Unit tests (see below) in at least the above browsers.
+
+If all the tests are working fine, you can finally test the extension versions, like this:
+
+  - Remove the '-WIP' from the version key from the manifest.json file present in the root of this repo;
+  - In Chromium, you can install the extension by loading the root folder with Extensions -> Load Unpacked (with Developer Mode turned ON) -> select the root folder of the repository;
+  - In Firefox, you can load an extension with Manage Your Extensions -> Debug Add-ons -> Load Temporary Add-on, and then pick any file in the repository.
+
+If your feature works and tests are passing, make a PR, describe the testing you have done, and ask for a code review.
+
+Please note that the app caches its own code so that it can run as an offline-first Progressive Web App. This can complicate development, because you
+may not see your changes, even after you refresh the browser. In Configuration, under "Expert settings", you will find a button that allows you to do
+a full app reset, which will erase the PWA. When Service Worker mode is turned on, there is also a checkbox that bypasses the App Cache. You can turn
+this on if you are frequently changing code and refreshing. Remember to turn it off for final testing. You can manually delete the App Cache in
```

Comment:
@Jaifroid I can see this option to bypass the cache in the jQuery mode too under the performance settings. Is there anything I am missing here?

Reply:
It hasn't been merged yet. You should be able to see it in this implementation:

https://kiwix.github.io/kiwix-js/

Open that, and switch to Service Worker mode. You should see the option. If you don't see it, check that the version number is 3.3.7 (top left of app). If it isn't 3.3.7, make sure you are in SW mode, and restart the app. You might have to do this more than once.

Reply:
Ah, I misunderstood your question. The option under Performance settings is a different option. That's simply an option not to cache assets. The other cache I'm referring to is the App Cache, which is relevant only for developers who might need to clear it while testing code. The app cache caches the app's own code, not assets.

Reply:
Ah, got it. So, will it be better to move that kind of option in some sort of developer mode instead (in future releases ofc)?

Reply:
We've put it under "Expert settings", and coloured it with a "warning" colour, so hopefully end users won't touch. We could look into hiding it behind a "dev mode" link, but for now, as there are only two options there, and one of them could actually be useful to end users whose app gets stuck on a prior version (the "Reset App" button), we should probably leave them exposed.

Reply:
Sure, makes sense.